safe-modules/liveness-module-v2.md

```diff
@@ -0,0 +1,139 @@
+the specific issues being faced by customers, and any constraints on a solution as defined either
+by customers or by technical limitations. Context and Problem Statement is an opportunity to tell
+the story that helps to motivate the rest of the design document. -->
+
+[Safe Accounts](https://safe.global) are configured as a set of owner addresses where some
+threshold of addresses can sign and ultimately execute a transaction through the account. We must
+solve for two safety cases, first that a threshold of signers do not maliciously execute a
+transaction (safety failure) and second that a blocking minority of signers
+(`total - threshold + 1`) don't become unable to sign (liveness failure). Either case can be
+equally catastrophic but we tend to focus on the safety failure case. Here we attempt to produce a
+solution for the liveness failure case that fits within certain constraints.
+
+We define multisig liveness as the ability for the multisig to reach consensus and execute a
+transaction within a bounded amount of time. Note that a multisig does not need to reach consensus
+over every possible transaction (because the members have the ability to reject transactions as
+needed). We instead say that a multisig is live as long as it is able to reach consensus over a
+transaction that states "this multisig is live".
+
+At a high level, a multisig fails to be live if at least a blocking minority of signers prevents
+the multisig from being able to reach consensus. This can happen if signers are honest but unable
+to sign (e.g., keys are lost) or if signers are malicious and refuse to sign. We must address both
+problems if we want to avoid multisig liveness failures. The existing `LivenessGuard` is
+insufficient in this regard because it does not handle malicious signers.
+
+Any system to address the above must also scale sufficiently such that a signer is not
+unnecessarily burdened by the liveness mechanism (this is a product constraint). The
+`LivenessGuard` is also insufficient here because it does not scale across multiple chains and it
+demands signatures on a regular basis (as opposed to only demanding signatures as needed).
+
+## Customer Description
+
+<!-- Provide a brief summary of the customers for this design document. -->
+
+The customers for this design doc are any participants in the multisig accounts that would utilize
+this module, as well as any 3rd parties who rely on the security properties of these accounts.
+
+### Customer Reviewers
+
+<!-- Identify at least one customer who should be involved in the review of this document. -->
+
+- Coinbase (potentially impacted 3rd party)
+- Uniswap (`LivenessModule` user)
+- Optimism Foundation (`LivenessModule` user)
+
+## Requirements and Constraints
+
+<!-- Identify the solution requirements and any additional design constraints from the Context and
+Problem Statement section in a bulleted list. -->
+
+- Solve both the owner liveness and account liveness problems *securely*.
```

Comment:
Maybe expand a bit on this one on what *securely* means. Which is not allowing malicious users to use the solution to maintain or increase their control of the multisig.